src/appservice/MjolnirManager.ts

```diff
@@ -70,6 +72,18 @@ export class MjolnirManager {
         await managedMjolnir.start();
         this.mjolnirs.set(mxid, managedMjolnir);
         this.unstartedMjolnirs.delete(mxid);
+        // @ts-ignore
+        if (this.instanceCountGauge._getValue({ status: "offline", uuid: localPart })) {
+            this.instanceCountGauge.dec({ status: "offline", uuid: localPart });
+        }
+        // @ts-ignore
+        if (this.instanceCountGauge._getValue({ status: "disabled", uuid: localPart })) {
+            this.instanceCountGauge.dec({ status: "disabled", uuid: localPart });
+        }
+        // @ts-ignore
+        if (!this.instanceCountGauge._getValue({ status: "online", uuid: localPart })) {
+            this.instanceCountGauge.inc({ status: "online", uuid: localPart });
+        }
```

Comment:
Not that we can't accommodate for this, but what is going on here?

Reply:
so the `_getValue` is private in the gauge. One way would be to cache it on the draupnir side again. But it seems rather weird to have 2 variables saving the same state here to me. So I went with evil mode here.

I could see why it might make sense to rewrite it though. (Tbh its a little weird to me why getValue is private in the upstream dependency. But its not like we can change that easily)

Reply:
Ok, that's ok. If it makes sense to you, i'd probably localise the private access to something like an instanceCountGuage to a function/method, e.g. `getGuageValue`, keeping the ts-ignore within that function and then add some context in comments for why it's necessary. What do you think?

Reply:
Sounds like a good idea. I will have a look at that on the weekend